File: server/router/business/memDepart.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type MemDepartRouter struct {
}

// InitMemDepartRouter 初始化 MemDepart 路由信息
func (s *MemDepartRouter) InitMemDepartRouter(Router *gin.RouterGroup) {
	router := Router.Group("memDepart").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("memDepart")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.MemDepartApi
	{
		router.POST("createMemDepart", apiV1.CreateMemDepart)   // 新建MemDepart
		router.POST("deleteMemDepart", apiV1.DeleteMemDepart) // 删除MemDepart
		router.POST("deleteMemDepartByIds", apiV1.DeleteMemDepartByIds) // 批量删除MemDepart
		router.POST("updateMemDepart", apiV1.UpdateMemDepart)    // 更新MemDepart
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findMemDepart", apiV1.FindMemDepart)        // 根据ID获取MemDepart
		routerNoRecord.GET("getMemDepartList", apiV1.GetMemDepartList)  // 获取MemDepart列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel MemDepart列表
	}
}

File: web/src/api/memDepart.ts
import { http,resData } from "@/utils/http";
 
// @Tags MemDepart
// @Summary 创建MemDepart
export const createMemDepart = (data?: object) => {
   return http.request<resData>("post", "/memDepart/createMemDepart", { data });  
}

// @Tags MemDepart
// @Summary 删除MemDepart
export const deleteMemDepart = (data?: object) => {
   return http.request<resData>("post", "/memDepart/deleteMemDepart", { data });  
}
 

// @Tags MemDepart
// @Summary 删除ByIdsMemDepart
export const deleteMemDepartByIds = (data?: object) => {
   return http.request<resData>("post", "/memDepart/deleteMemDepartByIds", { data });  
}
 

// @Tags MemDepart
// @Summary 更新MemDepart
export const updateMemDepart = (data?: object) => {
   return http.request<resData>("post", "/memDepart/updateMemDepart", { data });  
}

// @Tags MemDepart
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/memDepart/quickEdit", { data });  
}
 

// @Tags MemDepart
// @Summary 用id查询MemDepart
export const findMemDepart = (params?: object) => {
   return http.request<resData>("get", "/memDepart/findMemDepart", { params });  
}
 

// @Tags MemDepart
// @Summary 分页获取MemDepart列表
export const getMemDepartList = (params?: object) => {
   return http.request<resData>("get", "/memDepart/getMemDepartList", { params });  
}
 

// @Tags MemDepart
// @Summary 导出excelMemDepart列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/memDepart/excelList", { params });  
}
